### ISSUE TITLE
Fix S1144 FP: Record method PrintMembers

### ISSUE THREAD

**Issue by joepb** [SonarSource/sonar-dotnet#7904]
### Description

The PrintMembers(..) method on a sealed record class is incorrectly flagged as unused with csharpsquid:S1144.

### Repro steps

```cs
public sealed record Example 
{
    private bool PrintMembers(StringBuilder builder)   // Noncompliant - FP
    {
        return true;
    }
}
```

### Expected behavior

The code is valid and should not be flagged. The PrintMembers method is used by the runtime to create a string representation of the record.

See also the [documentation on records and PrintMembers](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record#printmembers-formatting-in-derived-records).

### Actual behavior

The method is flagged with warning S1144: Remove the unused private method 'PrintMembers'.

### Related information

* Enterprise Edition Version 9.9.1 (build 69595)
* SonarScanner for MSBuild 5.13.1
* Plugin: csharp version 8.51.0.59060
* Plugin: securitycsharpfrontend version 9.9.0.19083

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you for submitting this issue @joepb. Confirmed as False Positive.
